Theme1/src/firsttheme/PalindromeCheck.java

```diff
@@ -0,0 +1,38 @@
+package firsttheme;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PalindromeCheck {
+    private static final Pattern pattern = Pattern.compile("\\W");
```

Comment:
regexp for excluding whitespace looks redundant here. You're using `strings.split(" ")` which make a trim operation for each substring, so you will never get whitespace in this case